api/agent.go

```diff
@@ -483,3 +483,68 @@ type HostDataResponse struct {
 	AgentID  string
 	HostData *HostData `json:",omitempty"`
 }
+
+// GetSchedulerWorkerConfig returns the targeted agent's worker pool configuration
+func (a *Agent) GetSchedulerWorkerConfig() (*SchedulerWorkerPoolArgs, error) {
+	var resp AgentSchedulerWorkerConfigResponse
+	_, err := a.client.query("/v1/agent/schedulers/config", &resp, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &SchedulerWorkerPoolArgs{NumSchedulers: resp.NumSchedulers, EnabledSchedulers: resp.EnabledSchedulers}, nil
+}
+
+// SetSchedulerWorkerConfig attempts to update the targeted agent's worker pool configuration
+func (a *Agent) SetSchedulerWorkerConfig(args SchedulerWorkerPoolArgs) (*SchedulerWorkerPoolArgs, error) {
```

Comment:
I think we need `WriteOptions` here (and `QueryOptions` for `GetSchedulerConfig` above) to support ACLs and any HTTP params we might want in the future. And we always seem to want it eventually so that way we don't have to make a `SetSchedulerWorkerConfigWithOptions` later on.

We can probably get away without having a `QueryMeta` returned here because everything in `QueryMeta` is used for comes out of raft? None of the other agent APIs in this file have it.